scripts/valhalla_build_extract

```diff
@@ -342,27 +379,35 @@ if __name__ == '__main__':
     # override with inline-config
     config.update(**json.loads(args.inline_config))
 
-    # get and validate the tiles directory
+    # if output file exists and not --overwrite specified, fail
+    tiles_extract_out = args.extract_tar or config["mjolnir"].get("tile_extract")
+    if not tiles_extract_out:
+        LOGGER.critical(f"No output file path specified in 'mjolnir.tile_extract' or --extract-tar.")
+        sys.exit(1)
+    elif Path(tiles_extract_out).is_file() and not args.overwrite:
+        LOGGER.critical(f"File exists. Specify --overwrite to overwrite {tiles_extract_out}")
+        sys.exit(1)
+
+    # prefer the tar file to extract from, fall back to the tile dir
     tiles_dir: Path = Path(config["mjolnir"].get("tile_dir", '/dev/null'))
-    if not tiles_dir.is_dir():
+    # optionally use the tile extract to find the tiles
+    if args.extract_tar and (tiles_extract_in := Path(config["mjolnir"].get("tile_extract"))).is_file():
```

Comment:
first time ever I used the cute "walrus" operator haha it's python3.8 (3.7 is EOL by now)